src/modules/flutterwave/flutterwave.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { CreateFlutterwavePaymentDto } from './dto/create-flutterwavePaymentDto';
+import { CreatePaymentDto } from './dto/create-paymentDto';
+import { ConfigService } from '@nestjs/config';
+import { CustomHttpException } from '../../helpers/custom-http-filter';
+import { v4 as uuid4 } from 'uuid';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Payment, PaymentStatus } from './entities/payment.entity';
+
+@Injectable()
+export class FlutterwaveService {
+  private readonly secretkey: string;
+  private readonly baseUrl: string;
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly configService: ConfigService,
+    @InjectRepository(Payment)
+    private readonly paymentRepo: Repository<Payment>
+  ) {
+    this.secretkey = configService.get<string>('FLUTTERWAVE_SECRET_KEY');
+    this.baseUrl = configService.get<string>('FLUTTERWAVE_BASE_URL');
+  }
+
+  async initiatePayment(createFlutterwavePaymentDto: CreateFlutterwavePaymentDto): Promise<any> {
+    const headers = {
+      Authorization: `Bearer ${this.secretkey}`,
+      'Content-Type': 'application/json',
+    };
+    const payment_plan = await this.httpService
+      .get(`${this.baseUrl}/payment-plans/${createFlutterwavePaymentDto.plan_id}`, { headers })
+      .toPromise();
+    if (!payment_plan) {
+      return new CustomHttpException('Payment plan not found', 404).getResponse();
+    }
+    const paymentData = {
+      tx_ref: uuid4(),
+      amount: payment_plan.data.data.amount,
+      currency: payment_plan.data.data.currency,
+      redirect_url: createFlutterwavePaymentDto.redirect_url,
+      customer: {
+        email: createFlutterwavePaymentDto.email,
+        name: createFlutterwavePaymentDto.full_name,
```

Comment:
fetch full_name as `${first_name} ${last_name}` from userService.getUserRecord(), using the req.param.id